.github/workflows/run-all-wagon-tests.yml

```diff
@@ -0,0 +1,86 @@
+name: 'Run the tests in all wagons with a specific core revision'
+
+on:
+  workflow_dispatch:
+    inputs:
+      core_ref:
+        description: Use a specific version of the core for the workflow run. Defaults to master.
+        default: ""
+      wagon_dependency_ref:
+        description: Use a specific version of the wagon dependency for the workflow run. Defaults to master.
+        default: ""
+      wagon_ref:
+        description: Use a specific version of the wagon dependency for the workflow run. Defaults to master.
+        default: "master"
+  push:
+    branches: [ master ]
+    paths-ignore:
+      - 'doc/**'
+      - '**.md'
+  pull_request:
+    paths-ignore:
+      - 'doc/**'
+      - '**.md'
+  secrets:
+    WAGON_TESTS_GITHUB_TOKEN:
+      description: 'Token with repo and actions:read and actions:write for running the tests in all wagon repos'
+      required: true
+
+jobs:
+  find-wagons:
+    name: 'Find all Puzzle-managed wagons'
+    runs-on: 'ubuntu-latest'
+    outputs:
+      wagons: ${{ steps.list-wagon-repositories.outputs.wagons }}
+
+    steps:
+      - name: 'List wagon repositories'
+        id: list-wagon-repositories
+        run: |
+          WAGONS=$(gh repo list hitobito -L 100 --no-archived --json owner,name | \
+            jq 'map(select(.name | test("^hitobito_"))) | map({ owner: .owner.login, repo: .name, name: .name | match("^hitobito_(.+)$").captures[0].string })')
+          echo "wagons=$WAGONS" | tr -d "\n" >> $GITHUB_OUTPUT
```

Comment:
For easier debuggability and for new devs trying to understand what's going on, maybe echo the result normally as well